src/app/InteractionModelEngine.cpp

```diff
@@ -381,17 +452,36 @@ Protocols::InteractionModel::Status InteractionModelEngine::OnReadInitialRequest
             {
                 TLV::TLVReader pathReader;
                 eventpathListParser.GetReader(&pathReader);
-                err = TLV::Utilities::Count(pathReader, requestedEventPathCount, false);
+                TLV::Utilities::Count(pathReader, requestedEventPathCount, false);
             }
-            else if (err == CHIP_ERROR_END_OF_TLV)
+            else if (err != CHIP_ERROR_END_OF_TLV)
             {
-                err = CHIP_NO_ERROR;
+                return Status::InvalidAction;
             }
-            if (err != CHIP_NO_ERROR)
+
+            if (requestedAttributePathCount == 0 && requestedEventPathCount == 0)
             {
+                ChipLogError(InteractionModel,
+                             "Subscription from [%u:" ChipLogFormatX64 "] has no attribute or event paths. Rejecting request.",
+                             apExchangeContext->GetSessionHandle()->GetFabricIndex(),
+                             ChipLogValueX64(apExchangeContext->GetSessionHandle()->AsSecureSession()->GetPeerNodeId()));
                 return Status::InvalidAction;
             }
 
+            //
+            // TODO: We don't have an easy way to do a similar 'path expansion' for events to deduce
+            // access so for now, assume that the presence of any path in the event list means they
+            // might have some access to those events.
+            //
+            if (!hasValidAttributePath && requestedEventPathCount == 0)
+            {
+                ChipLogError(InteractionModel,
+                             "Subscription from [%u:" ChipLogFormatX64 "] has no access at all. Rejecting request.",
+                             apExchangeContext->GetSessionHandle()->GetFabricIndex(),
+                             ChipLogValueX64(apExchangeContext->GetSessionHandle()->AsSecureSession()->GetPeerNodeId()));
+                return Status::UnsupportedAccess;
```

Comment:
Again, this is not the right error status for this situation, per spec.

Reply:
InvalidAction?